### PR TITLE
Fresh Underground Preppers: LMOE shelter updates (part 1)

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Content "Updates and expands LMOE shelters."```

#### Purpose of change
LMOE shelters should be a very interesting thing to find, but really they're just pretty decent living spaces. They bear almost no resemblance to a real life private bunker, which is really more of an underground RV with thick steel walls. They lack a lot of newer furniture that would be appropriate as well. Plus they should be a place where the player is very likely to find NPCs.

#### Describe the solution
1. Update current LMOE shelters to be "empty" shelters that have been looted or lived in. These are the kind a player might start in. 
2. Add kitchens, generators, air and water filters, and pump wells rather than spring water to shelters. Hide valuables like guns behind walls instead of in the open. Reduce large areas of unused space.
3. Add ramps around the stone walls of the surface entrance and just generally punch it up a little.
4. Add a roof to the surface entrance
5. Add a new layout or two based on commercial bomb shelters
6. Add variants for all LMOE shelter maps: not empty. These have a very high chance of spawning with either survivors and a moderate amount of great loot, or deadly traps and a real treasure trove of loot. The survivors might be friendly... Or they might be very unhappy to see someone barging into their secret bunker after the end of the world.

Stretch goal: consider splitting existing lmoe shelter rooms into nested mapgen chunks so that there can be random variation of details.

LMOE1: Old and busted
LMOE1 old
LMOE1: New hotness - item spawns a bit too high currently
LMOE1 new

LMOE2: Old and really busted
LMOE2 old
LMOE2: New hotness - item spawns a bit too high currently
LMOE2 new

LMOE3: all new, based on a commercial bunker design. Still needs some fixes where the fill_ter is showing through.
LMOE3

New exterior: A hillock with the shelter concealed within
image

And there's even dirt on top of the dang thing.
image

Side note - ramps look awful.

#### Additional context
Why do we call them "LMOE shelters"? As far as I can tell nobody uses that term anywhere else...

### CODE REVIEW

**Comment by Maddremor** [comment]
Sounds good. 
Tangentially related: https://github.com/TheMurderUnicorn/PrepperCache

**Comment by Amoebka** [comment]
> "empty" shelters that have been looted or lived in. These are the kind a player might start in

Why would the shelters players start in be empty though? The "sheltered" scenario description makes it sound like the _food_ supplies are running out, but otherwise the shelter was properly stocked with tools/books/weapons/etc, and the survivor has never left it before to expend any of the above. Plus the entire point of the scenario is to have a quicker start in a harsher (more evolved) world.

**Comment by I-am-Erk** [comment]
Mainly because the player shelter needs to be picked dry of food and expendables, and I don't want to take the time to make a shelter type that has reusable tools and none of the food, matches, firewood, batteries, fuel... that's quite a bit of work for something seen only rarely and requires making a bunch of custom item groups.

**Comment by I-am-Erk** [comment]
I basically wound up completely remaking the second LMOE shelter map, it had tons of dead space rooms and empty hallways which doesn't make much sense if you're carving a shelter out of living rock.

Now it is a 4 bed shelter with a little kitchen, distinguished from the first by having more beds in a smaller space and being made from scrap metal and stone instead of being a totally enclosed steel shelter.

**Comment by Fris0uman** [comment]
So the new, better designed, shelter replace the previous one but they come in two flavor one empty and available and one with loot and NPC or Trap ?

Also could the old design be move to be low budget shelter, Like it was cheaper but also poorly design ? Or would it just be useless bloat ?

**Comment by Shodan14** [comment]
> Why do we call them "LMOE shelters"? As far as I can tell nobody uses that term anywhere else...

Not sure where it came from, but it means Last Man On Earth.

**Comment by I-am-Erk** [comment]
@Fris0uman 
The shelters really come in three flavours: "empty", which could be looted, or not stocked in the first place, or inhabited with its supplies used up; "inhabited", with NPCs living in it using some of the loot, but fully stocked; or "treasure trove", a fully stocked LMOE shelter with nobody using it. Treasure troves will have other sort of dangers though.

The old shelters *are* the new shelters. LMOE2 just didn't make much sense before... there's a ton of empty space there, dug out and reinforced presumably at massive expense to the original owner. LMOE1 is basically still the same, just polished off with new furniture and terrain to make it seem more real. I'm not going to be keeping the pre-upgrade maps, they were changed for good reasons. I would like to make bomb shelters that aren't LMOEs some day, made cheaper and less ideally.

@Shodan14 I'm aware what it stands for, I'm just not sure why we call them that and not "bomb shelter" or similar.

**Comment by Amneiger** [comment]
It looks like there's a room in the new LMOE 2 that doesn't have the door.  It's the room in the upper right hand corner with the handgun in it. 

I'm definitely looking forward to finding LMOEs with NPCs in them.  If the standard survivor can make it to the evac shelter, someone would have made it to their LMOE.   This will be even better once we finally get faction ownership of items/furniture/vehicles, or a way for NPCs to tell you to leave and then let you leave peacefully as long as you don't step off the stairs

**Comment by I-am-Erk** [comment]
> It looks like there's a room in the new LMOE 2 that doesn't have a door. It's the room in the upper right hand corner with the handgun in it.
> 
> I'm definitely looking forward to finding LMOEs with NPCs in them. If the standard survivor can make it to the evac shelter, someone would have made it to their LMOE. This will be even better once we finally get faction ownership of items/furniture/vehicles, or a way for NPCs to tell you to leave and then let you leave peacefully as long as you don't step off the stairs

That doorless room is intentional, it's just a little utility nook. It would be annoying to have to open and close a door to access the well pump.

**Comment by Rivet-the-Zombie** [comment]
>That doorless room is intentional, it's just a little utility nook. It would be annoying to have to open and close a door to access the well pump.

It doesn't appear to have an open doorway or anything; it's entirely sealed.

**Comment by I-am-Erk** [comment]
Oh, I misunderstood which one you meant. The one with the guns does in fact have a door, but it's not an obvious one.

**Comment by Night-Pryanik** [comment]
изображение
I'd replace this chainlink fence with a solid metal wall. I think there's absolutely no need for possible visitors to see what's beside this fence and farther.

By the way, is this a rain (in SW corner) underground?

**Comment by I-am-Erk** [comment]
I left the fence in partly as a legacy reference to the old map, and partly because LMOE2 represents a much more amateur design. The chain link is to make sure there's extra airflow from the generator out, which is why it's so close to the exit.

When I spawned the test maps I did it on z=1

**Comment by Mcrone617** [comment]
Is a lack of water pump intentional in the 3rd design? Sorry if i'm missing it, but it seems to be a very high quality vault, so it should have a renewable water pump right?

**Comment by I-am-Erk** [comment]
In my current version, not screenshotted, it does have one... But real commercial shelters don't have water pumps as far as I can find, they have internal water storage. I might just fill the storage tanks with a huge amount of clean water.

**Comment by Mcrone617** [comment]
> In my current version, not screenshotted, it does have one... But real commercial shelters don't have water pumps as far as I can find, they have internal water storage. I might just fill the storage tanks with a huge amount of clean water.

Yeah, that is a good point, but in game logic, a player is just going to dig one right near the entrance in the long run anyways, so for convenience I would keep pumps in the design just so you have to run around less. You could make the pump above ground, and refill your tanks every two weeks or so, but it's just more button presses at the end of the day.

**Comment by I-am-Erk** [comment]
In game button press logic isn't really part of designing a shelter to feel real.

**Comment by Mcrone617** [comment]
> In game button press logic isn't really part of designing a shelter to feel real.

Yeah fair enough, I did some basic looking into it, and came to the same conclusion you did. If nuclear blasts have contaminated the water supply, you don't want to be pumping it into your shelter anyways. Easier to wait it out with massive stores of water and hope by the time you run dry it's relatively safe on the surface. 
In game it's pretty easy to make your own pump and refill the tanks anyways, so it shouldn't be that much of a hassle.

**Comment by I-am-Erk** [comment]
Exactly... and the commercial shelter will have a huge supply of water to begin with. I'm debating adding the same to the professionally designed non-commercial one, possibly in addition to well water.

**Comment by chriseich21** [comment]
LMOE Shelter always made me think of the bunker in Lost, not sure how well it fits intentions/world, but may be inspiration for a possible variant or ideas.

**Comment by thethunderhawk** [comment]
I’m interested in that hidden gun room on LMOE2. It’s a secret door, yeah? If so, how does the meta-uninformed player discover it? Maybe if there were a noise coming from the room, or a note somewhere hinting at it it might be more rewarding to a new player’s curiosity.

**Comment by Throwaway-name** [comment]
No decontamination showers at the entrances?

File: data/json/overmap_terrain_military.json [hunk 1]
```diff
@@ -100,6 +109,15 @@
     "see_cost": 2,
     "flags": [ "KNOWN_UP", "NO_ROTATE" ]
   },
+  {
+    "type": "overmap_terrain",
+    "id": "lmoe_under_empty",
+    "name": "LMOE shelter",
+    "sym": 43,
```
Comment (kevingranade) [inline data/json/overmap_terrain_military.json]:
```suggestion
    "sym": "+",
```

Comment (I-am-Erk) [reply data/json/overmap_terrain_military.json]: Is that what Travis didn't like?

The error message seems quite vague if that's the case.

**Pull request merged**